### ISSUE TITLE
Missing artwork in several albums

### ISSUE THREAD

**Issue by rootsandculture** [squentin/gmusicbrowser#184]
I'm using the GTK3 build, and noticed many album are missing the covers. I launched gmusicbrowser from terminal and seems to be a problem with the character encoding or something related:

`no picture found in /home/rootsandculture/Music/M/-M- - Lettre Infinie (2019) [MP3] {3Ã¨me Bureau}/01 - Lettre Infinie.mp3
`
`no lyrics found in /home/rootsandculture/Music/1-10/7naRoda - ConvocaÃ§Ã£o (2018) [MP3] {7naRoda}/08 - Sagrado Respeito.mp3
`
In previous gmsuicbrowser versions just appears fine the same albums.

**Comment by squentin** [comment]
I haven't seen any problems like that with gtk3.
Can you check with the gtk2 version from the v1.1.x branch, the save file is compatible so you can go back and forth between the 2 versions with no issue.
Also can you try to set the cover manually and see if it works, and, even better, can you check if the line in the save file is the same before and after.

**Comment by rootsandculture** [comment]
The gtk2 version works ok, the artwork appears fine

`no lyrics found in /home/rootsandculture/Music/1-10/7naRoda - Convocação (2018) [MP3] {7naRoda}/05 - Vem Pra Ficar.mp3
`

Seems to be a character encoding or something like that, cause the error appears when uses some "strange" characters.

**Comment by rootsandculture** [comment]
I take a screenshot to explain it better, on the same config file, many albums doesn't show its covers on Gtk3 version:

![Screenshot from 2020-10-06 14-21-00](https://user-images.githubusercontent.com/71974607/95238300-b32b6880-07df-11eb-9146-ee2112aeff94.png)
![Screenshot from 2020-10-06 14-23-28](https://user-images.githubusercontent.com/71974607/95238306-b45c9580-07df-11eb-96ae-1e335546943b.png)

**Comment by squentin** [comment]
You didn't answer to "Can you try to set the cover manually and see if it works", as in right click on one of the missing cover, choose "set picture", and select the picture. Check if works, and check if it still work after quitting and restarting. It's important to know where the problem could be.
I trust me, I know about character encoding issues, I even keep a few folder and files in my own collection with non-utf8 encoding, which causes warnings or worse with many programs, just to check for any issue and it's working fine with them.

**Comment by rootsandculture** [comment]
> I tried manually adding the Artwork or selecting multiple albums by "auto select pictures" but still missing the pictures. Seems to be a character encoding or something like that, cause the error appears when uses some "strange" characters.

Yes I tried but the problem remains

**Comment by squentin** [comment]
> > I tried manually adding the Artwork or selecting multiple albums by "auto select pictures" but still missing the pictures. Seems to be a character encoding or something like that, cause the error appears when uses some "strange" characters.
> 
> Yes I tried but the problem remains
Ok but I need more details.
When you bring up the picture selection dialog:
Does it show the correct folder (the one that contains the cover picture) ?
If not can you go into that folder ?
Does it list the picture file ?
If it does, when you select it, does the picture show up in the preview ?
And I assume when you press ok, it doesn't show up in the cover mosaic ?

**Comment by rootsandculture** [comment]
>  Does it show the correct folder (the one that contains the cover picture) ?
Yes, I can select by the correct folder and the Folder.jpg but it doesn't shows up. I noticed in the terminal some strange characters from the covers than doesn't shows. That's why I assume it's character encoding related issue.
I has the issues with ElementaryOS based on 18.04LTS. Recently moved onto Ubuntu 20.04 and everything works fine. The only issue is than GTK3 version is much slower when search songs or artists, when press propierties from a file, etc. All the interface looks nice and modern but is slower.

**Comment by squentin** [comment]
Ok then that could be a bug that has been fixed upstream, unless someone see that bug on a more recent distribution I think I'll just ignore it. People using 18.04 can use v1.1.16 anyway (you can go back on forth between the gtk2 and gtk3 version), and currently it mostly has the same features.

The slowness is probably the SongList, it is slow when adding or removing songs, and I doubt I'll be able to fix it. But the SongTree is fast. I'll probably try to replace the SongList by the SongTree seamlessly. In the meantime you can edit your layout to use a SongTree.

**Comment by rootsandculture** [comment]
Yes, if I select layout with browser (SongTree) ir runs a lot faster. I wonder if can I achieve that speed with layout List, Library & Context?

**Comment by davidfoerster** [comment]
I experience the same issue – also on Ubuntu 18.04.

**Comment by morgannunan** [comment]
I've noticed this a lot, too, with different use cases. It would be nice if gmb could cache cover art because it looks like it's always reading directly from the files.

**Comment by davidfoerster** [comment]
@morgannunan As far as I can tell, GMB *does* cache cover art. Once images are loaded from a drive, a later spin-down doesn’t cause delay – at least not always. I don’t know the caching strategy here; it could very well be that images may be evicted from cache at some point under the assumption that it is relatively cheap to re-read and re-decode them later.

**Comment by davidfoerster** [comment]
@squentin Since I appear to experience the same issue I’ll try to answer your (to date) unanswered questions:

> Can you try to set the cover manually and see if it works.
Tried it. Still no cover picture.

> When you bring up the picture selection dialog:
> Does it show the correct folder (the one that contains the cover picture) ?
Yes.

> If not can you go into that folder ?
Yes, but not necessary.

> Does it list the picture file ?
Yes.

> If it does, when you select it, does the picture show up in the preview ?
No. (Works for non-affected albums.)

> And I assume when you press ok, it doesn't show up in the cover mosaic ?
Precisely.

**Comment by Ezekiel000** [comment]
I was having a problem with album art not showing when I updated to the gtk3 version. It turned out that all the album art not showing was in jpeg progressive format, so when I converted all of them to jpeg baseliine they all show up now. Could this be related?

**Comment by morgannunan** [comment]
I also don't think progressive vs baseline is the issue. @davidfoerster, can you elaborate on your discovery with regard to ASCII characters? My directory structure is /home/myuser/Music/M/Mp3 Name.mp3 so I don't think what you're describing is the issue I'm having. It is probably what OP was describing, though, and there's no fixing it. It doesn't matter if I manually replace the image. For me, the only solution was to downgrade gmusicbrowser from gmusicbrowser_1.1.99.1+1516 to gmusicbrowser_1.1.99.1+1514, and this broke my MPRIS integration so the album art is missing in the notification controls, half the time. I wrote about it in [the forum](https://forum.gmusicbrowser.org/index.php?topic=78168.0).

**Comment by morgannunan** [comment]
OK, I fixed the MPRIS issue, and I posted the solution in the forum, but I'll cross-post it here:

+ Install gmusicbrowser_1.1.99.1+1516;
+ Go to /usr/share/gmusicbrowser/plugins, and copy [mpris2.pm](https://mpris2.pm/) into a temp directory;
+ Downgrade to gmusicbrowser_1.1.99.1+1514;
+ copy [mpris2.pm](https://mpris2.pm/) from your temp directory into /usr/share/gmusicbrowser/plugins, overwriting the version that's there.

Now, I have all my covers showing, and they show in the indicator (so on my phone, the lock screen and pull down menu). As of this writing, the website links to 1514, and the nightly PPA contains 1516.

**Comment by davidfoerster** [comment]
> I also don't think progressive vs baseline is the issue. @davidfoerster, can you elaborate on your discovery with regard to ASCII characters? My directory structure is /home/myuser/Music/M/Mp3 Name.mp3 so I don't think what you're describing is the issue I'm having. It is probably what OP was describing, though, and there's no fixing it. It doesn't matter if I manually replace the image. For me, the only solution was to downgrade gmusicbrowser from gmusicbrowser_1.1.99.1+1516 to gmusicbrowser_1.1.99.1+1514, and this broke my MPRIS integration so the album art is missing in the notification controls, half the time. I wrote about it in [the forum](https://forum.gmusicbrowser.org/index.php?topic=78168.0).

I think we’re experiencing and describing two separate issues then. Mine fits OP’s exact symptoms but they only and reliably occur on directories with names that contain “special” characters. Extended Latin and most Unicode punctuation seem to work fine but certain Unicode characters don’t.

P. S.: I can’t identify any such character any longer. Maybe the problem was fixed in some underlying library because I don’t recall updating GMB.
